utils/color.go

```diff
@@ -0,0 +1,71 @@
+package utils
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"sync"
+
+	"github.com/ava-labs/avalanchego/utils/logging"
+)
+
+var supportedColors = []logging.Color{
+	logging.Cyan,
+	logging.LightBlue,
+	logging.LightGray,
+	logging.LightGreen,
+	logging.LightPurple,
+	logging.LightCyan,
+	logging.Purple,
+	logging.Yellow,
+}
+
+type color struct {
+	lock       sync.Mutex
+	usedColors []logging.Color
+}
+
+// ColorPicker allows to assign a color to different clients
+var ColorPicker = &color{
```

Comment:
Let's not use a global variable here. We could have a function `GetNextColor(i int)` which returns the color at `supportedColors[i%len(supportedColors)`. (That might be the easiest.) Or we could have a function `NewColorPicker` which returns a color picker struct.

Reply:
I think the use of a global variable is justified here. Specifically, the suggested change significantly deteriorates usage for clients. Also this is a fairly uncritical part of the code.